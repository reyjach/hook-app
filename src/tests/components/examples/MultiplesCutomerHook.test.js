import { shallow } from "enzyme";
import { MultiplesCutomerHook } from "../../../components/examples/MultiplesCutomerHook";
import { useCounter } from "../../../hooks/useCounter";
import { useFech } from "../../../hooks/useFech";

jest.mock("../../../hooks/useFech");
jest.mock("../../../hooks/useCounter");

describe('probar en el componente MultiplesCutomerHook', () => { 

    
    beforeEach(() => {
        useCounter.mockReturnValue({
            counter: 8,
            increment: () =>{},
        });
    });

    test('Debe de mostrarse correctamente', () => { 

        useFech.mockReturnValue({
            data:null,
            loading: true,
            error: null
        });

        const wrapper = shallow( <MultiplesCutomerHook /> );

        expect( wrapper ).toMatchSnapshot();

     });

     test('Debe de mostrar la informacion', () => { 

        useFech.mockReturnValue({
            data:[{
                author:'Jesica',
                quote:'hola mundo'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow( <MultiplesCutomerHook /> );

        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-0').text().trim() ).toBe( 'hola mundo' );
        expect( wrapper.find('footer').text().trim() ).toBe( 'Jesica' );


      })
 })