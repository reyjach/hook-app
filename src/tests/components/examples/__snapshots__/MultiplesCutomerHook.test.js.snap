// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probar en el componente MultiplesCutomerHook Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    The Breaking Bad
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
